app/grandchallenge/workstations/static/workstations/js/session-control.js

```diff
@@ -15,8 +15,14 @@ function openWorkstationSession(element) {
         }
 
         const regions = JSON.parse(document.getElementById('workstation-regions').textContent);
-        const potentialSessionOrigins = getSessionOrigins(domain, regions);
 
+        let potentialSessionOrigins;
+        if (domain.includes("localhost")) {
+            const port = window.location.port;
+            potentialSessionOrigins = Array(`http://${domain}${port ? ':' + port : ''}`);
+        } else{
+            potentialSessionOrigins = getSessionOrigins(domain, regions);
+        }
```

Comment:
Why isn't this handled in `getSessionOrigins`. Checking if the domain contains localhost does not seem the best way to go about this.

Reply:
Yes, I also don't like it. Maybe we could instead add a data-debug setting on the button that is only set in the tests and use that.